### DISCUSSION TITLE
How do i get the underlying interleaved buffer(view)?

### DISCUSSION THREAD

**Discussion by christjt** [donmccurdy/glTF-Transform#350] [Q&A] [answered]
Hey,

I am trying to use this lib to load a .glb model for a web-viewer. The model uses the "EXT_mesh_gpu_instancing" extension and contains nodes that links to some instanced attributes that are structured in an interleaved buffer defined by some bufferView. When using the `accessor.getArray()` method I only seem to load the typed array for that specific attribute. How do I go about fetching the underlying buffer that containes all my interleaved data. I need this because the buffer should be uploaded to the GPU as an interleaved buffer (not with each attribute in a separate buffer).

Any help would be appreciated . Thanks!

**Comment by donmccurdy** [comment, marked as answer]
Hi! Because this project is primarily meant for enabling modifications to glTF data, it does de-interleave accessors during the loading process. This allows changes to individual accessors to be simpler and faster, rather than maintaining a strict buffer layout at each stage of the processing. Accessors can (optionally) be re-interleaved when writing back to a glTF file.

If you'd like to interleave the attributes without exporting the whole file, you can find example code for that here:

https://github.com/donmccurdy/glTF-Transform/blob/0b244d44f9bb4504d8b5323d481ac0a6a979718b/packages/core/src/io/writer.ts#L93-L168

**Reply by christjt** [reply]
Thanks for the reply! Yeah I guess the main use case for this lib isn't being a generic loader, but I found that it is so well done and fantastic to use that I prefer it to other js/ts gltf loaders out there. And as a bonus, if I ever want to transform my gltfs pre-parsing, that would be very easy to do with this lib. 

In addition I also found that this lib was a little bit faster than f.ex threejs's gltf loader even though this lib de-interleaves the data (which was unexpected) when parsing large amount of data (~1000 glb files).

In any case, great job on this. Really awesome and intuitive to use, and thanks for the code snippet.
